src/services/listS3Objects.spec.ts

```diff
@@ -0,0 +1,63 @@
+import { mockClient } from 'aws-sdk-client-mock'
+import { listS3Objects } from './listS3Objects'
+import {
+  S3Client,
+  ListObjectsV2Command,
+  ListObjectsV2CommandInput
+} from '@aws-sdk/client-s3'
+
+const s3Mock = mockClient(S3Client)
+
+describe('list S3 objects', () => {
+  const input: ListObjectsV2CommandInput = {
+    Bucket: 'example-bucket'
+  }
+
+  beforeEach(() => {
+    s3Mock.reset()
+  })
+
+  test('response has no continuation token', async () => {
+    s3Mock.on(ListObjectsV2Command).resolves({
+      Contents: [{ Key: 'example-object' }]
+    })
+
+    const result = await listS3Objects(input)
+    expect(result).toEqual(['example-object'])
+  })
+
+  test('has continuation token', async () => {
```

Comment:
Could we make these slightly more descriptive e.g. Calls with continuation tokens return results for all pages

Reply:
Have done so, can you let me know if it's descriptive enough?

`response has continuation token - return results for all pages`